torch_xla/experimental/pjrt.py

```diff
@@ -259,57 +259,44 @@ def broadcast_master_param(model: nn.Module) -> None:
   xm.mark_step()
 
 
-def rendezvous(tag: str,
-               payload: bytes,
-               ordinals: Optional[List[int]],
-               *,
-               mesh_port: int = 12355) -> List[bytes]:
+def rendezvous(tag: str, payload: bytes,
+               ordinals: Optional[List[int]]) -> List[bytes]:
   """Share `payload` with all replicas in `ordinals`.
 
   All of PjRt is experimental right now, but consider `rendezvous` to be _very_
   experimental. Only tested on TPU v4.
 
   `tag` is ignored except for logging.
 
-  If `torch.distributed group` is not created already, `rendezvous` will
-  initialize it using `XRT_MESH_SERVICE_ADDRESS` or `MASTER_ADDR`. If world size
-  is 1, initialize the process group automatically.
+  Uses XLA collective communication to communicate between replicas, so this
+  will sync the graph (`xm.mark_step`).
 
   Args:
     tag: Name of this rendezvous operation.
     payload: Payload to share with other replicas.
     ordinals: List of replicas participating in rendezvous.
-    mesh_port: Port of master `torch.distributed` process.
   Returns:
     List of bytes from other replicas.
   """
-  if not dist.is_initialized():
-    logging.warning(
-        'Default process group not initialized. Creating XLA process group...')
-    mesh_master = xu.getenv_as("XRT_MESH_SERVICE_ADDRESS", str)
-    if not mesh_master and device_type() == 'TPU':
-      mesh_master = f'{tpu.discover_master_worker_ip()}:{mesh_port}'
-
-    if mesh_master:
-      init_method = f'tcp://{mesh_master}'
-    elif global_device_count() == 1:
-      init_method = f'file://{tempfile.mktemp()}'
-    else:
-      init_method = None
-
-    import torch_xla.distributed.xla_backend
-    dist.init_process_group(
-        "xla",
-        init_method=init_method,
-        world_size=global_device_count(),
-        rank=global_ordinal())
+  assert not ordinals or len(
+      ordinals) == global_device_count(), 'Only global rendezvous is supported'
+  assert isinstance(payload, bytes)
+  data = torch.tensor(list(payload), dtype=torch.uint8)
```

Comment:
Perhaps we should first call `xm.mark_step()` at the beginning of this rendezvous function before this line to finish all execution of previous graphs? Otherwise, the collective ops below might be added as part of the HLO of earlier model training or inference execution and trigger a recompilation of the model (that could take several minutes).

Reply:
Good idea! Done.